test-site/src/components/MapPin.tsx

```diff
@@ -1,9 +1,48 @@
-import { PinComponent } from '@yext/search-ui-react';
+import { useEffect, useRef, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { Map, Popup, LngLatLike } from 'mapbox-gl';
+import { PinComponent, Coordinate } from '@yext/search-ui-react';
+import { Result } from '@yext/search-headless-react';
 import { Location } from '../pages/LocationsPage';
+import { MapPopup } from './MapPopup';
+
+interface MapPinProps {
+  result: Result<Location>
+  index: number
+  mapbox: Map
+}
+
+const transformToMapboxCoord = (coordinate: Coordinate): LngLatLike => ({
+  lng: coordinate.longitude,
+  lat: coordinate.latitude,
+});
+
+export const MapPin: PinComponent<Location> = (props: MapPinProps) => {
+  const popupRef = useRef(new Popup({ offset: 15 }));
+  const initialRender = useRef(true);
```

Comment:
what is `initialRender` for? The useEffect below is guarded by the field `active` so I think we could remove this?

Reply:
This prevents the code running onMount

Reply:
The code in the useEffect below is guarded by `active` field, which is set to false as the initial state value. without `initialRender`, I believe the code portion for adding a popup still wouldn't run on the first render / onMount until active is set to true through `onClick`.

Reply:
Oh I see. I removed `active` in a followup commit so I thought you meant both weren't needed. This is fixed